Mathlib/NumberTheory/EulerProduct/Basic.lean

```diff
@@ -0,0 +1,170 @@
+/-
+Copyright (c) 2023 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.NumberTheory.SmoothNumbers
+import Mathlib.Analysis.SpecificLimits.Normed
+import Mathlib.Analysis.Normed.Field.InfiniteSum
+
+/-!
+# Euler Products
+
+The main results in this file are `EulerProduct.euler_product`, which says that
+if `f : ℕ → F` is norm-summable, where `F` is a complete normed field and `f` is
+multiplicative on coprime arguments with `f 0 = 0`, then
```

Comment:
There are already general definitions of weakly / strongly multiplicative functions in mathlib, in the `ArithmeticFunction` file. (I'm not saying we absolutely have to use those constructs here, of course, but we should be on guard against accidental duplication of existing functionality.)

Reply:
It probably makes sense to add versions where `f` is of type `Nat.ArithmeticFunction`. I don't think I've seen a notion of completely multiplicative arithmetic functions in that file, though.

Reply:
> I don't think I've seen a notion of completely multiplicative arithmetic functions in that file, though

That's covered by the general MonoidHom classes (as you observed); it's the weakly multiplicative ones which are slightly verbose to write out directly, so picking up and using an existing definition might well shorten the code. (I'm not sure I like the design of the `ArithmeticFunction` class much, so if you don't like it either, feel free not to use it; I just thought I'd mention that it's there.)

Reply:
I was thinking of adding a version like `Nat.ArithmeticFunction.IsMultiplicative.eulerProduct` so that one could do `f_isMul.eulerProduct <proof of norm-summability>` (instead of refactoring what I have; I think it makes sense to have an unbundled version as well).

Reply:
I've added that now.